prime-router/src/main/kotlin/azure/BlobAccess.kt

```diff
@@ -149,21 +159,46 @@ class BlobAccess() : Logging {
         }
 
         /**
-         * Download the blob at the given [blobUrl]
+         * Download the blob at the given [blobUrl] as a ByteArray
          */
-        fun downloadBlob(blobUrl: String): ByteArray {
+        fun downloadBlobAsByteArray(blobUrl: String, retries: Int = getBlobDownloadRetry()): ByteArray {
             val stream = ByteArrayOutputStream()
             logger.debug("BlobAccess Starting download for blobUrl $blobUrl")
-            stream.use { getBlobClient(blobUrl).downloadStream(it) }
+            val options = DownloadRetryOptions().setMaxRetryRequests(retries)
+            stream.use { getBlobClient(blobUrl).downloadStreamWithResponse(
+                it,
+                null,
+                options,
+                null,
+                false,
+                null,
+                null
+            ) }
             logger.debug("BlobAccess Finished download for blobUrl $blobUrl")
             return stream.toByteArray()
         }
 
+        /**
+         * Download the blob at the given [blobUrl] as BinaryData
+         */
+        fun downloadBlobAsBinaryData(blobUrl: String,  retries: Int = getBlobDownloadRetry()): BinaryData {
```

Comment:
Maybe add a small test for this?

Reply:
This should be covered [here](https://github.com/CDCgov/prime-reportstream/blob/bd7ac9c1b41209413b581b8649f92285011d401b/prime-router/src/test/kotlin/azure/BlobAccessTests.kt#L243C76-L243C76).

Let me know if you had something else in mind or I'm missing something.

Reply:
Ah totally missed that, that seems fine